### PR TITLE
[SPARK-24069][R] Add array_min / array_max functions

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR proposes to add array_max and array_min in R side too.

array_max:

```r
df <- createDataFrame(cbind(model = rownames(mtcars), mtcars))
mutated <- mutate(df, v1 = create_array(df$gear, df$am, df$carb))
head(select(mutated, array_max(mutated$v1)))
```

```
  array_max(v1)
1             4
2             4
3             4
4             3
5             3
6             3
```

array_min:

```r
df <- createDataFrame(cbind(model = rownames(mtcars), mtcars))
mutated <- mutate(df, v1 = create_array(df$mpg, df$cyl, df$hp))
head(select(mutated, array_min(mutated$v1)))
```

```
  array_min(v1)
1             6
2             6
3             4
4             6
5             8
6             6
```

## How was this patch tested?

Unit tests were added in `R/pkg/tests/fulltests/test_sparkSQL.R` and manually tested. Documentation was manually built and verified.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @felixcheung, maybe, I will try to group more next time .. (2 ~ 4 functions ?)

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -206,6 +206,8 @@ NULL
 #' df <- createDataFrame(cbind(model = rownames(mtcars), mtcars))
 #' tmp <- mutate(df, v1 = create_array(df$mpg, df$cyl, df$hp))
 #' head(select(tmp, array_contains(tmp$v1, 21), size(tmp$v1)))
+#' head(select(tmp, array_max(tmp$v1)))
+#' head(select(tmp, array_min(tmp$v1)))
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: let's combine these into one line? I think multiple examples in the pattern of "head(select(...))" isn't super useful to be honest

**Review by felixcheung — APPROVED** [review]
LGTM!

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
Merged to master.

Thanks for reviewing this @felixcheung.

**Pull request closed without merging**